op-node/node/node.go

```diff
@@ -319,13 +350,53 @@ func (n *OpNode) initMetricsServer(ctx context.Context, cfg *Config) error {
 	}
 	n.log.Info("starting metrics server", "addr", cfg.Metrics.ListenAddr, "port", cfg.Metrics.ListenPort)
 	go func() {
-		if err := n.metrics.Serve(ctx, cfg.Metrics.ListenAddr, cfg.Metrics.ListenPort); err != nil {
+		if err := n.metrics.Serve(n.postResourcesCtx, cfg.Metrics.ListenAddr, cfg.Metrics.ListenPort); err != nil {
 			log.Crit("error starting metrics server", "err", err)
 		}
 	}()
 	return nil
 }
 
+func (n *OpNode) initHeartbeat(_ context.Context, cfg *Config) error {
+	if !cfg.Heartbeat.Enabled {
+		return nil
+	}
+	var peerID string
+	if cfg.P2P.Disabled() {
+		peerID = "disabled"
+	} else {
+		peerID = n.P2P().Host().ID().String()
+	}
+
+	payload := &heartbeat.Payload{
+		Version: version.Version,
+		Meta:    version.Meta,
+		Moniker: cfg.Heartbeat.Moniker,
+		PeerID:  peerID,
+		ChainID: cfg.Rollup.L2ChainID.Uint64(),
+	}
+
+	go func(url string) {
+		if err := heartbeat.Beat(n.resourcesCtx, n.log, url, payload); err != nil {
+			log.Error("heartbeat goroutine crashed", "err", err)
+		}
+	}(cfg.Heartbeat.URL)
+	return nil
+}
+
+func (n *OpNode) initPProf(_ context.Context, cfg *Config) error {
```

Comment:
This function always returns `nil`, can just remove the return value.

Reply:
Same as with the heartbeat, see other comment.